src/components/AutoComplete/AutoComplete.react.js

```diff
@@ -0,0 +1,195 @@
+   * will later be mapped to function
+   */
+  filter?: 'caseInsensitiveFilter' | 'caseSensitiveFilter' | 'defaultFilter' |
+    'fuzzyFilter' | 'levenshteinDistanceFilter' | 'noFilter',
+  /** Dash-assigned callback that gets fired when the input changes. */
+  fireEvent?: () => void,
+  /** The content to use for adding floating label element */
+  floatingLabelText?: Node,
+  /** If true, field receives the property width: 100% */
+  fullWidth?: boolean,
+  /** The hint content to display */
+  hintText?: Node,
+  /** Autocomplete ID */
+  id: string,
+  /** Override style for list */
+  listStyle?: Object,
+  /** The max number of search results to be shown. By default it shows
+   * all the items which matches filter */
+  maxSearchResults?: number,
+  /** Delay for closing time of the menu */
+  menuCloseDelay?: number,
+  /** Props to be passed to menu */
+  menuProps?: Object,
+  /** Override style for menu */
+  menuStyle?: Object,
+  /** Auto complete menu is open if true */
+  open?: boolean,
+  /** If true, the list item is showed when a focus event triggers */
+  openOnFocus?: boolean,
+  /** Props to be passed to popover */
+  popoverProps?: Object,
+  /** Text being input to auto complete */
+  searchText?: string,
+  /** Dash callback to update props on the server. */
+  setProps?: () => void,
+  /** Override the inline-styles of the root element */
+  style?: Object,
+  /** Origin for location of target */
+  targetOrigin?: {
+    vertical?: 'top' | 'center' | 'bottom',
+    horizontal?: 'left' | 'middle' | 'right',
+  },
+  /** Override the inline-styles of AutoComplete's TextField element */
+  textFieldStyle?: Object,
+};
+
+type State = {
+  searchText: string,
+}
+
+const defaultProps = {
+  anchorOrigin: {vertical: 'bottom', horizontal: 'left'},
+  animated: true,
+  dashCallbackDelay: 500,
+  dataSource: [],
+  dataSourceConfig: {text: 'text', value: 'value'},
+  disableFocusRipple: true,
+  errorStyle: {},
+  errorText: null,
+  filter: "defaultFilter",
+  fireEvent: () => {},
+  floatingLabelText: null,
+  fullWidth: false,
+  hintText: null,
+  listStyle: {},
+  menuCloseDelay: 300,
+  menuProps: {},
+  menuStyle: {},
+  open: false,
+  openOnFocus: false,
+  popoverProps: {},
+  searchText: "",
+  setProps: () => {},
+  style: {},
+  targetOrigin: {vertical: 'top', horizontal: 'left'},
+  textFieldStyle: {},
+};
+
+export default class AutoComplete extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {searchText: this.props.searchText};
+    this.updateTextProps = _.debounce(this._updateTextProps, this.props.dashCallbackDelay);
+  }
+
+  componentWillReceiveProps(nextProps: Props): void {
+    if (nextProps.searchText !== null && nextProps.searchText !== this.props.searchText) {
+      this.handleChange(nextProps.searchText, this.props.dataSource, {});
+    }
+  }
+
+  handleChange = (searchText: string, dataSource: Array, params: Object) => {
```

Comment:
Let's add docstrings to the functions since it is not immediately clear what they actually do.